### PR TITLE
Crw 2081 deprecated

### PR DESCRIPTION
**Pull request by SDawley**
https://issues.redhat.com/browse/CRW-2081

Moving language servers out of their own repo and into crw-images
updating build.sh and get-sources files for new location and to use the uploadAssets scripts to push/pull assets from GH Releases.

Made some updates to configbump and traefik so they didnt get left behind

### CODE REVIEW

File: codeready-workspaces-configbump/build/rhel.binary.build.sh [hunk 1]
```diff
@@ -43,16 +44,16 @@ while [[ "$#" -gt 0 ]]; do
   shift 1
 done
 
-#call rhel.Dockerfile.extract.asets
 ARCH="$(uname -m)"
 tarball="asset-configbump-${ARCH}.tar.gz"
 chmod +x ./build/dockerfiles/*.sh
 ./build/dockerfiles/rhel.Dockerfile.extract.assets.sh
 
+tarball="asset-${SYNC_REPO}-${ARCH}.tar.gz"
```
Comment (nickboldt) [inline codeready-workspaces-configbump/build/rhel.binary.build.sh:52]: I've just fixed a problem w/ this in another commit... see https://github.com/redhat-developer/codeready-workspaces-images/commit/eb95d46243d4d849e32417da248a31e1845832e1#diff-bcf01e1d261674fc93120e9ba45c30376085af52958189e1ac8b0d47071157a9R53

File: codeready-workspaces-configbump/build/rhel.binary.build.sh [hunk 2]
```diff
@@ -43,16 +44,16 @@ while [[ "$#" -gt 0 ]]; do
...
 if [[ ${UPLOAD_TO_GH} -eq 1 ]]; then
   # upload the binary to GH
   if [[ ! -x ./uploadAssetsToGHRelease.sh ]]; then 
       curl -sSLO "https://raw.githubusercontent.com/redhat-developer/codeready-workspaces/${MIDSTM_BRANCH}/product/uploadAssetsToGHRelease.sh" && chmod +x uploadAssetsToGHRelease.sh
   fi
-  ./uploadAssetsToGHRelease.sh --publish-assets -v "${CSV_VERSION}" -b "${MIDSTM_BRANCH}" -n ${GH_RELEASE_NAME} "${WORKSPACE}/${tarball}"
+  ./uploadAssetsToGHRelease.sh --publish-assets -v "${CSV_VERSION}" -b "${MIDSTM_BRANCH}" --asset-name "${SYNC_REPO}" "${tarball}"
```
Comment (nickboldt) [inline codeready-workspaces-configbump/build/rhel.binary.build.sh:58]: please rebase - this change is already in crw-2 branch

File: codeready-workspaces-imagepuller/build/scripts/sync.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
```
Comment (nickboldt) [inline codeready-workspaces-imagepuller/build/scripts/sync.sh:1]: this file is already in the repo. please rebase / pull commits.

File: codeready-workspaces-operator-bundle-generated/manifests/codeready-workspaces.csv.yaml [hunk 4]
```diff
@@ -1141,9 +1141,9 @@ spec:
                 - name: RELATED_IMAGE_codeready_workspaces_stacks_php_plugin_registry_image_GIXDCMYK
                   value: registry.redhat.io/codeready-workspaces/stacks-php-rhel8@sha256:477066a0dd29e041cc1935a645371cf700e29fb559f041da35ff53e309720823
                 - name: RELATED_IMAGE_codeready_workspaces_theia_endpoint_plugin_registry_image_GIXDCMYK
-                  value: registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8@sha256:cfaeafd71f87973ae6f0215f6bd99015698ef66ef684b45b47ed6bbc7084dd85
+                  value: registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8@sha256:45f37d28e96821fb031080b5ed1db43de6a8bdf2fd497d038be0c1a433ef750f
                 - name: RELATED_IMAGE_codeready_workspaces_theia_plugin_registry_image_GIXDCMYK
-                  value: registry.redhat.io/codeready-workspaces/theia-rhel8@sha256:35475d592eefe0d121962b2e16ae33bfb723b5dc280b25c686df0a02581297d3
+                  value: registry.redhat.io/codeready-workspaces/theia-rhel8@sha256:d8a7acb7d4a3f2770f4091e020eba1b2a23de5fedf64c58caad7ad038bebbeb2
```
Comment (nickboldt) [inline codeready-workspaces-operator-bundle-generated/manifests/codeready-workspaces.csv.yaml:1146]: why these changes?

File: codeready-workspaces-configbump/build/rhel.binary.build.sh [hunk 5]
```diff
@@ -16,15 +16,16 @@ set -e
...
 
 MIDSTM_BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null || echo "crw-2-rhel-8")
 if [[ ${MIDSTM_BRANCH} != "crw-"*"-rhel-"* ]]; then MIDSTM_BRANCH="crw-2-rhel-8"; fi
 
 usage () {
     echo "
-Usage:   $0 -v [CRW CSV_VERSION] [--noupload] [-b MIDSTM_BRANCH] [-ght GITHUB_TOKEN]
-Example: $0 -v 2.y.0 --noupload
+Usage:   $0 -v [CRW CSV_VERSION] [--noupload] [-b MIDSTM_BRANCH] [-ght GITHUB_TOKEN] -n [GITHUB_RELEASE_NAME]
+Example: $0 -v 2.y.0 --noupload -n configbump
```
Comment (nickboldt) [inline codeready-workspaces-configbump/build/rhel.binary.build.sh:28]: could use SYNC_REPO here

File: codeready-workspaces-configbump/build/rhel.binary.build.sh [hunk 6]
```diff
@@ -36,21 +37,23 @@ while [[ "$#" -gt 0 ]]; do
     '-v') CSV_VERSION="$2"; shift 1;;
     '-b') MIDSTM_BRANCH="$2"; shift 1;;
     '-ght') GITHUB_TOKEN="$2"; export GITHUB_TOKEN="${GITHUB_TOKEN}"; shift 1;;
-    '-n'|'--noupload') UPLOAD_TO_GH=0;;
+    '-n') GH_RELEASE_NAME="$2"; shift 1;;
```
Comment (nickboldt) [inline codeready-workspaces-configbump/build/rhel.binary.build.sh:40]: is this the same value as the hardcoded SYNC_REPO?

File: codeready-workspaces-configbump/build/rhel.binary.build.sh [hunk 7]
```diff
@@ -36,21 +37,23 @@ while [[ "$#" -gt 0 ]]; do
...
+    '--noupload') UPLOAD_TO_GH=0;;
     '--help'|'-h') usage;;
   esac
   shift 1
 done
 
-#call rhel.Dockerfile.extract.asets
 ARCH="$(uname -m)"
+tarball="asset-configbump-${ARCH}.tar.gz"
```
Comment (nickboldt) [inline codeready-workspaces-configbump/build/rhel.binary.build.sh:48]: you've added a flag for the -n `GH_RELEASE_NAME` (which is different than the ASSET_NAME in https://github.com/redhat-developer/codeready-workspaces/blob/crw-2-rhel-8/product/uploadAssetsToGHRelease.sh#L65 ... could use the same variable for consistency?)

But here we're still hardcoding "configbump" ?

File: codeready-workspaces-plugin-java8-openj9/get-sources.sh [hunk 8]
```diff
@@ -5,16 +5,27 @@ scratchFlag=""
 JOB_BRANCH=""
 doRhpkgContainerBuild=1
 forceBuild=0
-pullAssets=0
+PULL_ASSETS=0
+DELETE_ASSETS=0
+PUBLISH_ASSETS=0
 generateDockerfileLABELs=1
 targetFlag=""
+UPSTREAM_JOB_NAME="crw-deprecated_${JOB_BRANCH}" # eg., 2.4
```
Comment (nickboldt) [inline codeready-workspaces-plugin-java8-openj9/get-sources.sh:13]: there should be no more upstream job since we're no longer pulling assets from Jenkins.

File: codeready-workspaces-configbump/build/rhel.binary.build.sh (hunk 6 above)
Comment (SDawley) [reply codeready-workspaces-configbump/build/rhel.binary.build.sh:40]: The hardcoded SYNC_REPO is only in configbump and traefik, since I started with the language servers I ended up making up my own name, but I could change everything to be called SYNC_REPO instead

File: codeready-workspaces-configbump/build/rhel.binary.build.sh (hunk 7 above)
Comment (SDawley) [reply codeready-workspaces-configbump/build/rhel.binary.build.sh:48]: I could change both SYNC_REPO and GH_RELEASE_NAME to ASSET_NAME

**Comment by nickboldt** [comment]
replaced by #132
